aten/src/ATen/native/native_functions.yaml

```diff
@@ -6711,16 +6711,20 @@
 - func: to_dense(Tensor self, ScalarType? dtype=None) -> Tensor
   variants: method
 
+# In to_dense.masked, the masked argument appears before dtype to avoid ambiguity with to_dense
+- func: to_dense.masked(Tensor self, bool? masked, ScalarType? dtype=None) -> Tensor
```

Comment:
@pearu - can we add this as a kwarg-only (separate with ",*,") at the end of the signature? Otherwise we'll run into [forward compatibility](https://github.com/pytorch/pytorch/wiki/PyTorch's-Python-Frontend-Backward-and-Forward-Compatibility-Policy) issues since masked can be used positionally.

Reply:
@cpuhrsch I'd like to but this leads to the issue reported in https://pytorch.slack.com/archives/CFZ7V4DNZ/p1678140643591249 . If there is a better fix, I'd be happy to apply it.

Reply:
Fixed via https://github.com/pytorch/pytorch/pull/96095#discussion_r1135544478